src/components/ButtonWalletPredicate/ButtonWalletPredicate.tsx

```diff
@@ -0,0 +1,44 @@
+import type { ButtonProps } from '@chakra-ui/react'
+import { Button } from '@chakra-ui/react'
+import type { InterpolationOptions } from 'node-polyglot'
+import { useCallback } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+type ButtonWalletPredicateProps = {
+  isValidWallet: boolean
+  invalidWalletTranslation?: string | [string, InterpolationOptions]
+  onWalletNotConnectedClick?: () => void
+} & ButtonProps
+
+export const ButtonWalletPredicate = ({
+  isValidWallet,
```

Comment:
Note: We've discussed this one together with @NeOMakinG - not a fan of it, but can't really come up with any vernacular that's better either.
cc @woodenfurniture this relates to our notion of "wallet supports chain" meaning possibly 3 different things:

1. the HDWallet supports the chain (e.g `supportsBTC`)
2. the HDWallet supports multi-account (i.e `supportsBip44Accounts`)
3. the wallet supports the current combination of chain and account number at *runtime* (i.e MM multichain adapter does support non-EVM chains, but not having the snap means we effectively don't at runtime)